internal/commands/diagnose/diagnose.go

```
@@ -22,38 +22,48 @@ var (
 	authcheckTemplateFS embed.FS
 )
 
+var diagnostics = []struct {
+	check        func() (any, error)
+	templateName string
+	templateFS   embed.FS
+}{
+	{
+		authCheck,
+		authcheckTemplate,
+		authcheckTemplateFS,
+	},
+}
+
 // diagnoseCmd represents the diagnose command
 var diagnoseCmd = &cobra.Command{
 	Use:   "diagnose",
 	Short: "Run diagnostic checks.",
 	Long: `This command runs diagnostic checks for various settings and configurations
 to attempt to identify issues that could cause the agent to function improperly.`,
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		fmt.Print("Running diagnosis checks...\n\n")
-		runNetworkCheck()
+		for _, diagnostic := range diagnostics {
+			data, err := diagnostic.check()
+			if err != nil {
+				return err
+			}
+			t := template.Must(template.
+				New(diagnostic.templateName).
+				ParseFS(diagnostic.templateFS, diagnostic.templateName))
+			if err := t.ExecuteTemplate(os.Stdout, diagnostic.templateName, data); err != nil {
+				return err
```

if a specific check fails, we might not want to return the error and skip the rest of the checks. maybe it would be better to have some sort of default error message that we print out

---

Good idea! I added warning icons so it's more clear when there's a failure, but lmk if we're trying to stick to ASCII and I can remove those